src/toil/wdl/toilwdl.py

```diff
@@ -0,0 +1,2077 @@
+        '# set destination for java to deposit all of its files' + '\n' + \
+        'export _JAVA_OPTIONS=-Djava.io.tmpdir=$tmpDir' + '\n' + \
+        'export TMPDIR=$tmpDir' + '\n\n' + \
+        '(' + '\n' + \
+        'cd /' + docker_dir + '/execution' + '\n' + \
+        cmd + '\n' + \
+        ')' + '\n\n' + \
+        '# gather the input command return code' + '\n' + \
+        'echo $? > "$tmpDir/rc.tmp"' + '\n\n'
+
+    bashfile_string = bashfile_prefix
+
+    begin_globbing_string = \
+        '(' + '\n' + \
+        'cd $tmpDir' + '\n' + \
+        'mkdir "$tmpDir/globs"' + '\n'
+
+    bashfile_string = bashfile_string + begin_globbing_string
+
+    for glob_input in globs:
+        add_this_glob = \
+            '( ln -L ' + glob_input + \
+            ' "$tmpDir/globs" 2> /dev/null ) || ( ln ' + glob_input + \
+            ' "$tmpDir/globs" )' + '\n'
+        bashfile_string = bashfile_string + add_this_glob
+
+    bashfile_suffix = \
+        ')' + '\n\n' + \
+        '# flush RAM to disk' + '\n' + \
+        'sync' + '\n\n' + \
+        'mv "$tmpDir/rc.tmp" "$tmpDir/rc"'
+
+    bashfile_string = bashfile_string + bashfile_suffix
+
+    with open(temp_dir + '/' + job_name + '_script.sh', 'w') as bashfile:
+        bashfile.write(bashfile_string)
+    bashfile.close()
+
+class ToilWDL:
+    '''
+    A program to run WDL input files using native Toil scripts.
+
+    Currently in alpha testing, and known to work with the Broad's GATK tutorial
+    set for WDL on their main wdl site:
+    software.broadinstitute.org/wdl/documentation/topic?name=wdl-tutorials
+
+    And ENCODE's WDL workflow:
+    github.com/ENCODE-DCC/pipeline-container/blob/master/local-workflows/encode_mapping_workflow.wdl
+
+    Additional support to be broadened to include more features soon.
+    '''
+
+    def __init__(self, wdl_filename, secondary_filename, output_directory):
+
+        # inputs
+        self.wdl_file = wdl_filename
+        self.secondary_file = secondary_filename
+        self.output_directory = output_directory
+
+        if not os.path.exists(self.output_directory):
+            try:
+                os.makedirs(self.output_directory)
+            except:
+                raise OSError(
+                    'Could not create directory.  Insufficient permissions or disk space most likely.')
+
+
+        self.output_file = os.path.join(self.output_directory, 'toilwdl_compiled.py')
+
+        # only json is required; tsv/csv are optional
+        self.json_dict = {}
+        self.tsv_dict = {}
+        self.csv_dict = {}
+
+        # holds task skeletons from WDL task objects
+        self.tasks_dictionary = {}
+
+        # holds workflow structure from WDL workflow objects
+        self.workflows_dictionary = {}
+
+        # unique iterator to add to cmd names
+        self.command_number = 0
+
+        # unique number for a job
+        self.task_number = 0
+
+        # a job's 'level' on the DAG
+        self.task_priority = 0
+
```

Comment:
module_string and main_prelude string should be class attributes

Reply:
I've made module_string into its own def in case this wants to be handled dynamically in the future and incorporated the other as a def supporting the main write_main() def.